tests/src/org.eclipse.jgit/org.eclipse.jgit/6.5.0.202303070854-r/build.gradle

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.eclipse.jgit:org.eclipse.jgit:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+    testImplementation 'ch.qos.logback:logback-classic:1.4.5'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "metadata-user-code-filter.json"
+                extraFilterPath = "metadata-extra-filter.json"
```

Comment:
Why is this needed? Why does `userCodeFilterPath` alone not work?

Reply:
The `userCodeFilterPath` filters metadata entries by `typeReachable` class. The `extraFilterPath` filters by the metadata `name` class. Without `extraFilterPath` the following entry would end up in the generated metadata file
```
    {
        "name": "java.security.SecureRandomParameters",
        "condition": {
            "typeReachable": "org.eclipse.jgit.lib.Repository"
        }
    }
```
Since entries like that one are not needed, `extraFilterPath` is used.

Reply:
Thanks for the explanation!